### ISSUE TITLE
[FR] settings setup guide

### ISSUE THREAD

**Issue by matmair** [inventree/InvenTree#2284]
**Problem**
New installs can be configured a bunch - some stuff (like base url, name etc.) should really be set.

**Suggested solution**
- [ ] A setup modal if it is the first run
- [ ] Multipage wizard that guides through the experience
- [ ] Mixin that enables plugins to show setup page
- [ ] API endpoint to check if setups ran -> could be important for 3rd party systems
- [ ] Maybe a role for users/admins to specify who gets shown setup screens

**Describe alternatives you've considered**
Make a whole setup guide in the docs - I do not think that would be as effective.

**Examples of other systems**
OctoPrint has a great setup workflow

**Do you want to develop this?**
The structure yes, I would like to discuss the details of what to show in the core contributer group

**Comment by matmair** [comment]
This is a vast one - deferring to later.

@SchrodingersGat @eeintech I think our Out of Box Experience (OOBE) for new settings and plugins (with settings) needs some improvement. This will take some time but should tackle it if we get more users to reduce the starting friction and support load.

**Comment by SchrodingersGat** [comment]
Great idea @matmair - I am seeing this as a separate feature to the [FTUE](https://github.com/inventree/InvenTree/issues/2907) FR?

**Comment by matmair** [comment]
> Great idea @matmair - I am seeing this as a separate feature to the [FTUE](https://github.com/inventree/InvenTree/issues/2907) FR?

@SchrodingersGat yes. I would see this to be much more important than #2907 and also as an architectural base to build up on for that.

**Comment by matmair** [comment]
See https://github.com/inventree/InvenTree/pull/10715, that does most of the checklist items
